packages/block-editor/src/components/block-list/index.js

```diff
@@ -119,8 +120,10 @@ function Root( { className, ...settings } ) {
 		},
 		settings
 	);
+
 	return (
 		<IntersectionObserver.Provider value={ intersectionObserver }>
+			<DisableNonMainBlocks />
```

Comment:
```suggestion
			<MaybeDisableNonMainBlocks />
```

We might want to consider naming this as per the condition. Currently it reads as though it would always disable main blocks.